src/main/java/core/basesyntax/service/impl/DataConverterServiceImpl.java

```diff
@@ -0,0 +1,58 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.DataConverterService;
+import java.util.Arrays;
+import java.util.List;
+import java.util.NoSuchElementException;
+import java.util.Set;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+
+public class DataConverterServiceImpl implements DataConverterService {
+    private static final String SPLIT_REGEX = ",";
+    private static final String NOT_FOUND_ERR = "No such operation found.";
+    private static final int FIRST_INDEX = 0;
+    private static final int SECOND_INDEX = 1;
+    private static final int THIRD_INDEX = 2;
+
+    @Override
+    public List<FruitTransaction> getListOfTransactions(List<String> lines) {
+        Predicate<String> startsWithValidCode = new Predicate<>() {
```

Comment:
`Predicate` is a functional interface, therefore you should use in a functional fashion rather than creating an anonymous class.